### PR TITLE
Provide Makefile targets for additional checks

### PR DESCRIPTION
**Pull request by okurz**
Having all usual checks available over the Makefile makes it easier for
developers to see what is available and call them.

Let me know what you think

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,12 +8,20 @@ update-deps:
...
 .PHONY: test-tidy
 test-tidy:
 	tools/tidyall --all --check-only
 
+.PHONY: test-critic
+test-critic:
+	tools/perlcritic --quiet .
+
+.PHONY: test-yaml
+test-yaml:
```
[the remark this answers is not in the document]
Comment (okurz) [reply Makefile:22]: sure, can do

[the remark this answers is not in the document]
Comment (perlpunk) [reply Makefile:22]: actually, not needed:
> yamllint auto-detects when it’s running inside of GitHub Actions and automatically uses the suited output format to decorate code with linting errors. 

 https://yamllint.readthedocs.io/en/latest/integration.html#integration-with-github-actions
so we can call `make test-yaml` in the workflow.

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+test-tidy:
+	tools/tidyall --all --check-only
+
+.PHONY: test-critic
+test-critic:
+	tools/perlcritic --quiet .
+
+.PHONY: test-yaml
+test-yaml:
+	yamllint -c .yamllint --strict ./
```
Comment (perlpunk) [inline Makefile:23]:
```suggestion
	yamllint --strict ./
```
https://yamllint.readthedocs.io/en/latest/configuration.html#configuration
> a file named .yamllint, .yamllint.yaml, or .yamllint.yml in the current working directory, or

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (okurz) [reply Makefile:22]: Even better :)

[the remark this answers is not in the document]
Comment (okurz) [reply Makefile:22]: Actually I didn't use the make call in the github CI definition as the container does not have make. But I just added that minor dependency in https://build.opensuse.org/package/show/home:okurz:container/yamllint and as soon as the new version of the container is published then this should work

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**